src/connections/allContacts.js

```diff
@@ -104,3 +102,18 @@ export const withContactsMutations = withMutations(client => ({
 }))
 
 export default flow([withContacts, withContactsMutations])
+
+export function mergeContact(contact, attributes) {
+  function concat(arr1 = [], arr2 = []) {
+    return [...arr1, ...arr2]
+  }
+  function customizer(objValue, srcValue, key) {
```

Comment:
What about making it top-level, so we could unit-test attribute merging logic on its own without having to deal with a full contact object in every test?

Although some people consider that *private* functions should not be exposed for testing purpose, maybe it's worth the trouble.

Array attributes would probably still need to be tested as a whole in order to be able to either merge two corresponding items or keep them separate depending on their content.

Reply:
I understand your proposal, however:
The added value of the functionality is the merge of contacts, a unit test on an implementation detail does not seem easy to maintain.

Reply:
@sebn I refactored the way I manage merge strategies. Can you take a look at it?